app/Http/Controllers/AlbumController.php

```diff
@@ -541,123 +550,183 @@ public function getArchive(Request $request)
+		$albumIDs = explode(',', $request['albumIDs']);
 
-					if ($UserId == 0 || $user->upload) {
-						$photos_sql = Photo::select_stars(Photo::OwnedBy($UserId));
-						break;
+		if (count($albumIDs) === 1) {
+			switch ($albumIDs[0]) {
+				case 'f':
+					$zipTitle = 'Starred';
+					break;
+				case 's':
+					$zipTitle = 'Public';
+					break;
+				case 'r':
+					$zipTitle = 'Recent';
+					break;
+				case '0':
+					$zipTitle = 'Unsorted';
+					break;
+				default:
+					$album = Album::find($albumIDs[0]);
+					if ($album === null) {
+						Logs::error(__METHOD__, __LINE__, 'Could not find specified album');
+
+						return 'false';
 					}
-				}
-				$photos_sql = Photo::select_stars(Photo::whereIn('album_id', $this->albumFunctions->getPublicAlbums()));
-				break;
-			case 's':
-				$zipTitle = 'Public';
-				$photos_sql = Photo::select_public(Photo::OwnedBy($UserId));
-				break;
-			case 'r':
-				$zipTitle = 'Recent';
-				if ($this->sessionFunctions->is_logged_in()) {
-					$user = User::find($UserId);
-
-					if ($UserId == 0 || $user->upload) {
-						$photos_sql = Photo::select_recent(Photo::OwnedBy($UserId));
-						break;
+					$zipTitle = str_replace($badChars, '', $album->title);
+					if ($zipTitle === '') {
+						$zipTitle = 'Untitled';
 					}
-				}
-				$photos_sql = Photo::select_recent(Photo::whereIn('album_id', $this->albumFunctions->getPublicAlbums()));
-				break;
-			case '0':
-				$zipTitle = 'Unsorted';
-				$photos_sql = Photo::select_unsorted(Photo::OwnedBy($UserId));
-				break;
-			default:
-				$album = Album::find($request['albumID']);
-				if ($album === null) {
-					Logs::error(__METHOD__, __LINE__, 'Could not find specified album');
-
-					return 'false';
-				}
-				$zipTitle = $album->title;
-
-				$photos_sql = Photo::set_order(Photo::where('album_id', '=', $request['albumID']));
-
-				Logs::notice(__METHOD__, __LINE__, $album->title . ' has been downloaded.');
-
-				//TODO: we do not provide pictures from sub albums but it would be a nice thing to do later...
-
-//				->orWhereIn('album_id',function ($query) { function ($query) use ($id)
-//				{
-//					$query->select('album_id')
-//						->from('user_album')
-//						->where('parent_id','=',$id);
-//				}}));
-				break;
+					break;
+			}
+		} else {
+			$zipTitle = 'Albums';
 		}
 
-		$zipTitle = str_replace($badChars, '', $zipTitle) . '.zip';
-
-		$response = new StreamedResponse(function () use ($zipTitle, $photos_sql, $badChars) {
-			$opt = array(
-				'largeFileSize' => 100 * 1024 * 1024,
-				'enableZip64' => true,
-				'send_headers' => true,
-			);
-
-			$zip = new ZipStream($zipTitle, $opt);
-
-			// Check if album empty
-			if ($photos_sql->count() == 0) {
-				Logs::error(__METHOD__, __LINE__, 'Could not create ZipStream without images');
-
-				return 'false';
-			}
+		$response = new StreamedResponse(function () use ($albumIDs, $badChars) {
+			$zip = new ZipStream(null);
+
+			$UserId = $this->sessionFunctions->id();
```

Comment:
what happens here if the user is not logged in ?

Reply:
```php
	/**
	 * Return the current ID of the user
	 * what happens when UserID is not set? :p.
	 *
	 * @return int
	 */
	public function id()	{
		return Session::get('UserID');
	}
```

Reply:
I believe we don't read the value unless somebody is logged on. This code was there before as well so clearly it works :smiley:.

Reply:
:laughing: